In [11]:
import pandas as pd

df = pd.read_csv('data/e-commerce.csv', encoding='latin-1')


In [12]:
print(df.shape)


(541909, 8)


#Answer 541909 rows and 8 columns



In [34]:
df.info()

<class 'pandas.DataFrame'>
RangeIndex: 541909 entries, 0 to 541908
Data columns (total 9 columns):
 #   Column       Non-Null Count   Dtype  
---  ------       --------------   -----  
 0   InvoiceNo    541909 non-null  str    
 1   StockCode    541909 non-null  str    
 2   Description  540455 non-null  str    
 3   Quantity     541909 non-null  int64  
 4   InvoiceDate  541909 non-null  str    
 5   UnitPrice    541909 non-null  float64
 6   CustomerID   406829 non-null  float64
 7   Country      541909 non-null  str    
 8   Revenue      541909 non-null  float64
dtypes: float64(3), int64(1), str(5)
memory usage: 37.2 MB


Customer ID in float has no meaning as it is a categorical variable
no nun-null values -> very clean
Invoice data should be used as pd.datetime idk why I have this initiutive

In [13]:
print(df.head())
#p

  InvoiceNo StockCode                          Description  Quantity  \
0    536365    85123A   WHITE HANGING HEART T-LIGHT HOLDER         6   
1    536365     71053                  WHITE METAL LANTERN         6   
2    536365    84406B       CREAM CUPID HEARTS COAT HANGER         8   
3    536365    84029G  KNITTED UNION FLAG HOT WATER BOTTLE         6   
4    536365    84029E       RED WOOLLY HOTTIE WHITE HEART.         6   

      InvoiceDate  UnitPrice  CustomerID         Country  
0  12/1/2010 8:26       2.55     17850.0  United Kingdom  
1  12/1/2010 8:26       3.39     17850.0  United Kingdom  
2  12/1/2010 8:26       2.75     17850.0  United Kingdom  
3  12/1/2010 8:26       3.39     17850.0  United Kingdom  
4  12/1/2010 8:26       3.39     17850.0  United Kingdom  


print the first 5 records index 0 to 4
what is invoice no 
what is inovice date
unitprice means the e.g the white metal lantern sold at 3.39 pounds
both invoice no. is the same as customer id?
the customer id 17850 has ordered these things

In [14]:
df[df['CustomerID'] == 17850]['InvoiceNo'].nunique()

35

In [15]:
df['CustomerID'].nunique()

4372

In [21]:
No_of_transactions_in_each_country=df.groupby('Country')['InvoiceNo'].nunique()
No_of_transactions_in_each_country

Country
Australia                  69
Austria                    19
Bahrain                     4
Belgium                   119
Brazil                      1
Canada                      6
Channel Islands            33
Cyprus                     20
Czech Republic              5
Denmark                    21
EIRE                      360
European Community          5
Finland                    48
France                    461
Germany                   603
Greece                      6
Hong Kong                  15
Iceland                     7
Israel                      9
Italy                      55
Japan                      28
Lebanon                     1
Lithuania                   4
Malta                      10
Netherlands               101
Norway                     40
Poland                     24
Portugal                   71
RSA                         1
Saudi Arabia                2
Singapore                  10
Spain                     105
Sweden                     46
Sw

In [26]:
No_of_transactions_in_each_country.idxmax()


'United Kingdom'

In [ ]:
df['Revenue']=df['Quantity']*df['UnitPrice']
#Revenue gain from each record

In [30]:
#total revenue

sum=df['Revenue'].sum()
sum

np.float64(9747747.934)

In [33]:
df.groupby('Description')['Quantity'].sum().idxmax()

'WORLD WAR 2 GLIDERS ASSTD DESIGNS'